### DISCUSSION TITLE
Installing CoolProp in Ubuntu: Externally-managed-environment error

### DISCUSSION THREAD

**Discussion by inamdarharshad** [CoolProp/CoolProp#2444] [Q&A] [answered]
I upgraded my Ubuntu installation from 22.04 LTS to 24.04 LTS. Then, I tried to install CoolProp using the following command from the terminal:  
`pip install CoolProp`  
  
I also attempted the following:  
`pip install CoolProp --upgrade`  
  
Both attempts resulted in the following error message: 

> error: externally-managed-environment
>
> × This environment is externally managed 
>╰─> To install Python packages system-wide, try apt install
> &emsp;&emsp;&ensp; python3-xyz, where xyz is the package you are trying to
> &emsp;&emsp;&ensp; install.

  
I searched for solutions and found recommendations on StackExchange to first create a virtual environment and then install CoolProp, as follows: 
`apt install pipx`  
`pipx install CoolProp`

I was wondering if someone else experienced the same and tried the virtual environment solution.

**Comment by ibell** [comment, marked as answer]
Yeah, it is annoying, but I can see why they did it to keep packages from clashing with standard libraries. I bit the bullet and use virtualenv on linux. Actually I use conda mostly.

**Reply by inamdarharshad** [reply]
Is CoolProp available as a package in Conda? I use the Anaconda distribution on Windows, but did not find CoolProp in the list of packages.

**Reply by ibell** [reply]
And older version is, but not the most up-to-date I believe. I usually use pip.
